text/0230-telemetry-viewer.md

```diff
@@ -0,0 +1,169 @@
+service. In addition, operators often find it challenging to understand
+OpenTelemetry configurations at the Collector or in an SDK. While OpAMP provides
+an API that can help with this, it doesn't provide a management plane or
+visualization components to see and modify configurations in the browser.
+
+To address these gaps, and others, I propose a new 'OpenTelemetry Viewer'
+component that can be built into a collector and provides in-memory storage,
+viewing, and modification of OpenTelemetry data and components.
+
+## Explanation
+
+Different users of a telemetry system have different needs and expectations for
+debugging instrumentation. Currently, developers have two options for
+quick feedback - using a debug exporter at the collector or SDK, or using an
+existing analysis tool (open source or proprietary). Both of these options have
+drawbacks - the debug exporter presents an overwhelming amount of text-based
+data, and depending the characteristics of a development environment, it may be
+challenging to stand up and use a local suite of open source analysis tools
+(such as Jaeger, Prometheus, and OpenSearch) or use a commercial tool. Existing
+options are optimized for viewing, querying, and analyzing data across hundreds
+or thousands of sources, not for understanding "what attributes are my metrics
+emitting?" or "what type of data are my instrumentation libraries emitting?".
+
+Reduced cycle time (both in a DevOps sense and also in a more general reading of
+the word) is a contributor to quality and resiliency of software and human
+systems. Being able to quickly get feedback about if your changes are having the
+desired effect or not is invaluable, especially for developers that are
+beginning to instrument their services for observability.
+
+As custom instrumentation and data transformation becomes a larger and larger
+part of the OpenTelemetry story, enabling these fast feedback loops is critical
+for the project. However, we must also balance this against other tools in the
+ecosystem. Thus, I proprose the following set of criteria that will guide the
+implementation of this OTEP:
+
+- To be consistent with our existing stance on vendor agnosticism, any component
+  developed cannot implement persistent storage. Storage must be local
+  (constrained to the machine where OTV is running or accesssed from), and can
+  only persist as part of a session (for example, between refreshes on a browser
+  page or collector restarts).
+- We will not implement a query language or semantics as part of this project.
+  Data can be filtered and projected, but without a bespoke or pre-built query
+  language.
+
+With that said, the requirements of OTV are as follows:
+
+- An in-memory data store constrained by size and time. For example, a ring
+  buffer. This store can persist to local disk for persistence through collector
+  reboots, but is specifically not designed for long-term storage of data.
```

Comment:
I think so, consider the following use case. I'm trying to do complex OTTL transforms. I see that it's not working as expected, so I change the statement and trigger a hot reload/reboot of the collector. On restart, the cached data is reloaded, so I can see the old/new statement (rather than only viewing the new one)